code-challenges.js

```diff
@@ -9,9 +9,33 @@ var people = [
   { name: "arthur dent", occupation: "radio employee" }
 ]
 // Expected output example: "Ford Prefect is a hitchhiker." "Zaphod Beeblebrox is a president of the galaxy." "Arthus Dent is a radio employee."
+// declare a function that takes array as argument
+const sentenceMaker = (array) => {
+  // ** looking back, I could have probably destructored the objects from the array **
+  // need to do this process for each object, so will have to map through the objects
+  return array.map(object => {
+    // split the string into it's own index in array
+    let splitNames = object.name.split(" ")
+    // console.log(`at index ${index}: ${splitNames}`)
+    // map through each subarray to capitalized the first letter of index
+    // basically go through all the words in the name (so map through 2 items in the name array)
+    let upperCaseName = splitNames.map(word => {
+      // make sure to split the word into letters
+      let splitWord = word.split("")
+      // console.log(splitWord)
+      // only capitalize the first letter of each word 
+      // reassign the word to be the uppercased first letter and the remaining letters after splitting at index 1 (joined)
+      return word = splitWord[0].toUpperCase() + splitWord.slice(1).join("")
+      // console.log(splitWord[0].toUpperCase())
+      // then join the name together
+    }).join(" ")
+      // concat the occupation and "is a" and period to form a sentence 
+      return upperCaseName.concat(" is a ").concat( object.occupation).concat(".")
+      // then return all the sentences by joining them (so we convert the array of sentences to a string of 3 sentences)
+  }).join(" ")
+}
 
-
-
+console.log(sentenceMaker(people))
 
```

Comment:
Nice job. Actually I don't think you can destructure the objects here. There is no nested object. It looks great.